### PR TITLE
[W12H02] added multithreaded tests

### PR DESCRIPTION
**Pull request by HendrikHuebner**

### CODE REVIEW

**Review by Anatoly03 — COMMENTED** [review]
Are system prints really a good idea in test code?

**Comment by HendrikHuebner** [comment]
> Are system prints really a good idea in test code?

I kinda forgot to remove those but they could be useful for debugging I guess

**Comment by MagnusSaurbier** [comment]
Shouldn't the declared package be pgdp.ds instead of just ds?

**Review by MagnusSaurbier — CHANGES_REQUESTED** [review]
Package declaration: ds-->pgdp.ds

File: w12h02/test/pgdp/ds/MultiStackSyncTests.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
+package ds;
```
[the remark this answers is not in the document]
Comment (HendrikHuebner) [reply w12h02/test/pgdp/ds/MultiStackSyncTests.java]: this has already been fixed

**Review by MagnusSaurbier — APPROVED** [review]
LGTM

**Comment by MagnusSaurbier** [comment]
Could you take a look at my class TimeoutChecker? This makes it possible to set a timeout, after which the tests will terminate and fail - instead of running on forever - even under deadlock conditions.
This might improve these tests.

File: w12h02/test/pgdp/ds/MultiStackSyncTests.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
+package pgdp.ds;
+
+import org.junit.jupiter.api.Test;
+import pgdp.ds.MultiStack;
```
Comment (felixhauptmann) [inline w12h02/test/pgdp/ds/MultiStackSyncTests.java:4]: This is obsolete now that you changed the package. (leads to compiler error)

```suggestion
```

**Review by felixhauptmann — CHANGES_REQUESTED** [review]
Apart from my suggestion LGTM

File: w12h02/test/pgdp/ds/MultiStackSyncTests.java (hunk 2 above)
Comment (Anatoly03) [reply w12h02/test/pgdp/ds/MultiStackSyncTests.java:4]: The way Guthub works this suggestion will overwrite package

**Pull request merged**